### PR TITLE
Feature/nex 1017/service oauth2 refactor

### PR DESCRIPTION
**Pull request by ekkinox**
See https://oat-sa.atlassian.net/browse/NEX-908

### CODE REVIEW

**Review by major66 — APPROVED** [review]

File: Tests/Resources/Kernel/config/routes.yaml [hunk 1]
```diff
@@ -1,3 +1,7 @@
-test_launch:
-    path: /test/launch
-    controller: OAT\Bundle\Lti1p3Bundle\Tests\Resources\Action\Launch\TestLaunchAction
+test_message:
+    path: /test/message
+    controller: OAT\Bundle\Lti1p3Bundle\Tests\Resources\Action\Tool\Message\TestMessageAction
+
+test_service:
+    path: /test/service
+    controller: OAT\Bundle\Lti1p3Bundle\Tests\Resources\Action\Platform\Service\TestServiceAction
```
Comment (emgolubev) [inline Tests/Resources/Kernel/config/routes.yaml]: I would add a line to the end of file

**Review by emgolubev — APPROVED** [review]
Good job, approved, just added a little optional comment

**Comment by emgolubev** [comment]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Code is self documented

File: Tests/Resources/Kernel/config/routes.yaml (hunk 1 above)
Comment (ekkinox) [reply Tests/Resources/Kernel/config/routes.yaml]: fixed

**Review by major66 — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Code is self documented

File: doc/configuration.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+            tool: "localTool"
+            deployment_ids:
+                - "deploymentId1"
+            platform_key_chain: "platformKey"
+            tool_key_chain: "toolKey"
+            platform_jwks_url: ~
+            tool_jwks_url: ~
+```
+
+In this setup, the bundle allow your application to **act as a platform and as a tool**.
```
Comment (m-lagarde) [inline doc/configuration.md]: the bundle allows :)

Comment (ekkinox) [reply doc/configuration.md]: fixed

**Pull request merged**